### ISSUE TITLE
[FEATURE] Quake-like Drop down Terminal using Pyprland term (+other Pyprland stuff [zoom])

### ISSUE THREAD

**Issue by Somnius** [mylinuxforwork/dotfiles#155]
**[FEATURE] Quake-like Drop down Terminal using Pyprland term (+other Pyprland stuff [zoom])**
I would like to check with @mylinuxforwork if he would like to add a dropdown terminal like `yaquake` / `guake` and others.
From what i've checked Hyprland [supports drop-down stuff since Feb'2023](https://github.com/hyprwm/Hyprland/issues/1510), that is why i searched for it! 😉

**Suggestion**
I would suggest to utilize [pyprland](https://github.com/hyprland-community/pyprland) scripts in order to make it work.

**Implementation**
It would be used with 
```
bind = $mainMod SHIFT, Return, exec, pypr toggle term   # Quake-like drop-down terminal
```

**Additional context**
I just like drop-down terminals 😁, and some stuff you can do with Pyprland scripts

but i'd suggest a zoom function from `pypr` using something like the following
```
bind = $mainMod SHIFT, Z, exec, pypr zoom   # Toggle Pyprland Zoom
```

**Comment by mylinuxforwork** [comment]
I will try it out.

**Comment by Somnius** [comment]
> I will try it out.

Did you have some time to play with it pyprland, it's fun! 😊

**Comment by dwilliam62** [comment]
I am going to 2nd the motion for this. ;)   I use the dropdown terminal very often.  I set the terminal binding to SUPER + ~ (tilde)   

I've attached some config files   I have tested this with ML4W also. 

[pyprland.toml.txt](https://github.com/user-attachments/files/18472446/pyprland.toml.txt)

[pypr.keybinds.txt](https://github.com/user-attachments/files/18472448/pypr.keybinds.txt)

![Image](https://github.com/user-attachments/assets/c4820a79-1831-4e6d-bac2-f683cdc1a303)

**Comment by dwilliam62** [comment]
One benefit of this over special workspace is it's always there.   And you can still use special workspace for something else.  One thing I use it for is doing a NIXOS rebuilld or update, or compile something I run it there.  Then no matter what workspace I am in I can recall it, to check the status

**Comment by dwilliam62** [comment]
FYI:  Screenshot with ML4W dotfiles

![Image](https://github.com/user-attachments/assets/71d07b40-175e-48c7-b550-5917a86df21e)

**Comment by mylinuxforwork** [comment]
The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?

**Comment by dwilliam62** [comment]
> The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?

Does that offer a terminal ?

**Comment by mylinuxforwork** [comment]
> > The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?
> 
> Does that offer a terminal ?

Absolutely. Terminal, Browser or Newelle configurations are shipped. But every app can run as sidepad.

**Comment by dwilliam62** [comment]
> > > The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?
> > 
> > 
> > Does that offer a terminal ?
> 
> Absolutely. Terminal, Browser or Newelle configurations are shipped. But every app can run as sidepad.

Thanks  I'll check it out

**Comment by mylinuxforwork** [comment]
> > > > The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?
> > > 
> > > 
> > > Does that offer a terminal ?
> > 
> > 
> > Absolutely. Terminal, Browser or Newelle configurations are shipped. But every app can run as sidepad.
> 
> Thanks I'll check it out

https://mylinuxforwork.github.io/dotfiles/usage/sidepad

**Comment by dwilliam62** [comment]
> > > > > The version 2.9.9.1 includes a sidepad feature. Maybe a good alternative for you?
> > > > 
> > > > 
> > > > Does that offer a terminal ?
> > > 
> > > 
> > > Absolutely. Terminal, Browser or Newelle configurations are shipped. But every app can run as sidepad.
> > 
> > 
> > Thanks I'll check it out
> 
> https://mylinuxforwork.github.io/dotfiles/usage/sidepad

Hmmmm   Not really no.  4 keybinds to deal with/remember?   That just seems way, way over complicated.  esp compared to pyprland.   One keybind to toggle.  I can have multiple windows or apps directly on keybinds.  I don't have to initialize, decide what I want it to be the open close it with different keybinds.  I can set the size, direction it comes from with a simple config file.   I'll stay with pyprland 

<img width="1760" height="954" alt="Image" src="https://github.com/user-attachments/assets/a6941b71-d681-4a97-b19a-aa150974be75" />

**Comment by mylinuxforwork** [comment]
Thanks for your feedback. It’s an experiment and needs more tweaking of course. And it’s optional.

**Comment by Somnius** [comment]
> I'll stay with pyprland

same here

> It’s an experiment and needs more tweaking of course

Eager to see more

**Comment by dwilliam62** [comment]
> Thanks for your feedback. It’s an experiment and needs more tweaking of course. And it’s optional.

Yup understood!   Just meant you might want to make it less complicated to use.  I.e more dedicated sidebar apps. A binding to bring out  the terminal.   Another binding to bring out an AI panel,  etc...  When I use my drop down I want it now. ;)   In nix if I am building something or compiling an app I will put it there then quickly recall it from wherever I happen to be.  Or for ssh sessions etc     Thanks

**Comment by dwilliam62** [comment]
I've been using pyprland dropdown a lot recently wanted to show you an example 

<img width="1442" height="808" alt="Image" src="https://github.com/user-attachments/assets/c7cb5e24-ffbf-41b7-a7c6-71019bd7298a" />

I use the tab feature of the terminal a lot.  And tmux running on remote so I can disconnect/reconnect later.  It's a simple keybind, simple to add more dropdowns and easy to size, and set location.  If that can be done with your sidebar fantastic!  Just seemed like more steps vs. pyprland
